packages/concerto-core/test/introspect/stringvalidator.js

```diff
@@ -87,48 +128,120 @@ describe('StringValidator', () => {
             }).should.throw(/Validator error for field `id`. org.acme.myField/);
         });
 
+        it('should validate a string', () => {
+            let v = new StringValidator(mockField, null, VALID_MIN_LENGTH_AND_MAX_LENGTH_AST);
+            v.validate('id', 'AB1234567455455455');
+        });
+
+        it('should validate a string, when only minLength specified', () => {
+            let v = new StringValidator(mockField, null, NO_MAX_LENGTH_AST);
+            v.validate('id', 'AB1234567455455455');
+            (() => {
+                v.validate('id', 'w');
+            }).should.throw(/The string length of w should be at least 2 characters./);
+            (() => {
+                v.validate('id', '');
+            }).should.throw(/Validator error for field `id`. org.acme.myField/);
+        });
+
+        it('should validate a string, when only maxLength specified', () => {
+            let v = new StringValidator(mockField, null, NO_MIN_LENGTH_AST);
+            v.validate('id', 'ABCD123456');
+            v.validate('id', '');
+            (() => {
+                v.validate('id', 'ABCD1234567');
+            }).should.throw(/The string length of ABCD1234567 should not exceed 10 characters./);
+        });
+
+        it('should string legth should take precedence over regex (if string length specified)', () => {
+            let v = new StringValidator(mockField, { pattern: '^[A-z]{1,100}$' }, { minLength : 1, maxLength : 10 });
+            (() => {
+                v.validate('id', 'AbCdefghijklmksadada');
+            }).should.throw(/The string length of AbCdefghijklmksadada should not exceed 10 characters./);
+        });
+
     });
 
     describe('#validate with custom RegEx engine', () => {
         const options = {
             regExp: XRegExp
         };
         it('should ignore a null string', () => {
-            let v = new StringValidator(mockField, { pattern: '^[A-z][A-z][0-9]{7}' }, options);
+            let v = new StringValidator(mockField, { pattern: '^[A-z][A-z][0-9]{7}' }, VALID_MIN_LENGTH_AND_MAX_LENGTH_AST, options);
             v.getRegex().toString().should.equal('/^[A-z][A-z][0-9]{7}/');
             v.validate('id', null);
         });
 
         it('should validate a string', () => {
-            let v = new StringValidator(mockField, { pattern: '^[\\p{Letter}\\p{Number}]{7}', flags: 'u' }, options);
+            let v = new StringValidator(mockField, { pattern: '^[\\p{Letter}\\p{Number}]{7}', flags: 'u' }, VALID_MIN_LENGTH_AND_MAX_LENGTH_AST, options);
```

Comment:
Why do we need to change the existing tests? It's important that if we skip this parameter, then we should get the same behaviour.

Reply:
I can't skip that param, for now passing null instead of valid length ast. 
Since here we are passing `options` as one more param, validator will try to bind it to length validator if we don't explicitly pass null/valid length validator